services/venice-controller/src/main/java/com/linkedin/venice/controller/util/AdminUtils.java

```diff
@@ -0,0 +1,113 @@
+    final Store store = admin.getStore(clusterName, storeName);
+    if (store == null) {
+      LOGGER.warn(
+          "No store found in the store repository. Will get store-level RMD version ID from cluster config. "
+              + "Store name: {}, cluster: {}",
+          storeName,
+          clusterName);
+    } else if (store.getRmdVersion() == ConfigConstants.UNSPECIFIED_REPLICATION_METADATA_VERSION) {
+      LOGGER.info("No store-level RMD version ID found for store {} in cluster {}", storeName, clusterName);
+    } else {
+      LOGGER.info(
+          "Found store-level RMD version ID {} for store {} in cluster {}",
+          store.getRmdVersion(),
+          storeName,
+          clusterName);
+      return store.getRmdVersion();
+    }
+
+    final VeniceControllerClusterConfig controllerClusterConfig =
+        admin.getMultiClusterConfigs().getControllerConfig(clusterName);
+    if (controllerClusterConfig == null) {
+      throw new VeniceException("No controller cluster config found for cluster " + clusterName);
+    }
+    final int rmdVersionID = controllerClusterConfig.getReplicationMetadataVersion();
+    LOGGER.info("Use RMD version ID {} for cluster {}", rmdVersionID, clusterName);
+    return rmdVersionID;
+  }
+
+  /**
+   * Check if a store can support incremental pushes based on other configs. The following rules define when incremental
+   * push is allowed:
+   * <ol>
+   *   <li>If the system is running in single-region mode, the store must by hybrid</li>
+   *   <li>If the system is running in multi-region mode,</li>
+   *   <ol type="i">
+   *     <li>Hybrid + Active-Active</li>
+   *     <li>Hybrid + !Active-Active + {@link DataReplicationPolicy} is {@link DataReplicationPolicy#AGGREGATE}</li>
+   *     <li>Hybrid + !Active-Active + {@link DataReplicationPolicy} is {@link DataReplicationPolicy#NONE}</li>
+   *   </ol>
+   * <ol/>
+   * @param multiRegion whether the system is running in multi-region mode
+   * @param hybridStoreConfig The hybrid store config after applying all updates
+   * @return {@code true} if incremental push is allowed, {@code false} otherwise
+   */
+  public static boolean isIncrementalPushSupported(
+      boolean multiRegion,
+      boolean activeActiveReplicationEnabled,
+      HybridStoreConfig hybridStoreConfig) {
+    // Only hybrid stores can support incremental push
+    if (!AdminUtils.isHybrid(hybridStoreConfig)) {
+      return false;
+    }
+
+    // If the system is running in multi-region mode, we need to validate the data replication policies
+    if (!multiRegion) {
+      return true;
+    }
+
+    // A/A can always support incremental push
+    if (activeActiveReplicationEnabled) {
+      return true;
+    }
+
+    DataReplicationPolicy dataReplicationPolicy = hybridStoreConfig.getDataReplicationPolicy();
+    return dataReplicationPolicy == DataReplicationPolicy.AGGREGATE
+        || dataReplicationPolicy == DataReplicationPolicy.NONE;
+  }
```

Comment:
Should we leverage this in the `UpdateStoreUtils` for validation of scenarios for incremental push allowed? Seems redudant checks

Reply:
Yeah, I'll update it